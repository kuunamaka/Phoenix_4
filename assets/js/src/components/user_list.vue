<template>
  <table>
    <h2>Listing Users</h2>
    <li v-for="user in orderedUsers" :key="user.id">
      id: <b>{{ user.id }}</b>,
      name: <b>{{ user.name }}</b>,
      username: <b>{{ user.username }}</b>
      <a :href="`/users/${ user.id }`" class="button">Edit</a>
    </li>
    <a href="/users/new" class="button">Add User</a>
    <a href="/tasks" class="button">Task</a>
  </table>
</template>

<script>
import _ from 'lodash'
import axios from 'axios'
export default {
  name: 'user_list',
  data() {
    return {
      name: '',
      username: '',
      users: null
    }
  },
  computed: {
    orderedUsers: function() {
      return _.orderBy(this.users, 'id')
    }
  },
  async mounted() {
    const resp = await axios.get('/api/users')
    this.users = resp.data.users
  }
}
</script>
