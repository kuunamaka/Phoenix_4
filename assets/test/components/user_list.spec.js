import { shallowMount } from '@vue/test-utils'
import _ from 'lodash'
import user_list from  '@/src/components/user_list.vue'

const axios = require("axios");
const MockAdapter = require("axios-mock-adapter");
const mock = new  MockAdapter(axios);

mock.onGet("/api/users").reply(200, {
  users: [
    {id: 1, name: "Maui", username: "Kuunamaka"},
    {id: 2, name: "Tasuku", username: "Kaleorani"}
  ]
});
axios.get("/api/users").then(function(response) {
  console.log(response.data.users)
});

window = Object.create(window);
const pathname = '/users';

Object.defineProperty(window, 'location', {
  value: {
    pathname,
    href: `http://localhost:4001${ pathname }`
  }
});

describe('user_list.vue', () => {
  const wrapper = shallowMount(user_list);

  it('is a Vue instance', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('order users by their id', () => {
    const expectUsers = [
      {id: 1, name: "Maui", username: "Kuunamaka"},
      {id: 2, name: "Tasuku", username: "Kaleorani"}
    ];
    const vm = wrapper.vm

    expect(vm.users).toEqual(expectUsers)
  })
});
