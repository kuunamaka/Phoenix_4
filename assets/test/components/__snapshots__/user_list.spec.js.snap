// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user_list.vue renders entire component correctly 1`] = `
<table>
  <h2>
    Listing Users
  </h2>
   
  <li>
    
    id: 
    <b>
      1
    </b>
    , name: 
    <b>
      Maui
    </b>
    , username: 
    <b>
      Kuunamaka
    </b>
     
    <a
      class="button"
      href="/users/1"
    >
      Edit
    </a>
  </li>
  <li>
    
    id: 
    <b>
      2
    </b>
    , name: 
    <b>
      Tasuku
    </b>
    , username: 
    <b>
      Kaleorani
    </b>
     
    <a
      class="button"
      href="/users/2"
    >
      Edit
    </a>
  </li>
   
  <a
    class="button"
    href="/users/new"
  >
    Add User
  </a>
   
  <a
    class="button"
    href="/tasks"
  >
    Task
  </a>
</table>
`;
